### DISCUSSION TITLE
Works with Mac?

### DISCUSSION THREAD

**Discussion by aspen33619** [PydPiper/pylightxl#78] [Q&A] [answered]
Hi,
I am new to Python and pylightxl. The first steps were very promising, but now, I am stuck. I have a Mac M1 with the latest Excel-Version. I use Python 3 with PyCharm. I can read XLSX-Files without problems, but when I try to update cells, the program simply does not do anything. No cell-update, no error message. Can that be caused by using the Mac or is it likely to be a coding issue?

Thank you for your time.

**Comment by PydPiper** [comment, marked as answer]
@aspen33619 thank you for considering using pylightxl. I wont be able to get back to you in detail for a few hours here but can you give a bit more description on each step that you took to see what could be causing the problem. Your code, double check file paths stuff like that

**Reply by aspen33619** [reply]
Thanks a lot for your immediate reply. I have attached my code. Of course, it is not very smart. First, I am not an experienced programmer and second, I have tried different approaches to the problem (i.e. address and row/col-notation). I have also tried to open a second object (dbw) for writing. Only using the db-object did not help either. Maybe, I have screwed the code completely in the meantime. Is there still anything that can help you to help me in that code?

Thanks again.

```
import requests
import pylightxl as xl
import subprocess

STARTROW = 2
COLUMN_IP = 'A'
COLUMN_PING = "B"
XL_LOCATION = "/Users/XXX/Documents/ShellyTest.xlsx"
SHEET_NAME = 'Shelly_YYY'
HTTP_PREFIX = 'http://'
STATUS_ANNEX_V1 = '/status'
STATUS_ANNEX_V2 = '/rpc/Shelly.GetStatus'

# https://pylightxl.readthedocs.io/en/latest/quickstart.html#read-excel-file
db = xl.readxl(fn=XL_LOCATION)


# Zähler festlegen in Initialwert für Vergleich bestimmen
CurrentLine = STARTROW
ShellyAddress = db.ws(ws=SHEET_NAME).address(address=COLUMN_IP + str(CurrentLine))

while ShellyAddress != '':
    ShellyAddress = db.ws(ws=SHEET_NAME).address(address=COLUMN_IP + str(CurrentLine))
    PingCell = COLUMN_PING + str(CurrentLine)
    print(PingCell)
    # ping
    res = subprocess.call(['ping', '-c', '3', ShellyAddress])
    if res == 0:
        dbw = xl.readxl(fn=XL_LOCATION)
        dbw.ws(ws=SHEET_NAME).update_index(row=CurrentLine, col=2, val=100)
        print("ping to", ShellyAddress, "OK")
    elif res == 2:
        dbw = xl.readxl(fn=XL_LOCATION)
        print("no response from", ShellyAddress)
        dbw.ws(ws=SHEET_NAME).update_address(address=PingCell, val=200)
        print('Written: Value 200 to address ',PingCell)
    else:
        print("ping to", ShellyAddress, "failed!")
        db.ws(ws=SHEET_NAME).update_address(address=PingCell, val=300)
    print("Ping Result:", str(res))

    RequestAddress = HTTP_PREFIX + ShellyAddress + STATUS_ANNEX_V2
    print('Adresse: ' + ShellyAddress)
    RequestURL = HTTP_PREFIX + ShellyAddress + STATUS_ANNEX_V2
    print(RequestURL)
    try:
        print('In Try')
        print('Result: ' + str(res))
        MyRequest = requests.get(RequestURL, timeout=2)
        print(MyRequest.status_code)
    except:
        print('In Except')
        # print(MyRequest.status_code

    CurrentLine = CurrentLine + 1
```

**Reply by PydPiper** [reply]
Oh okay, yeah so you never saved the updates. This library does not dynamically update (not streaming). You'll need to save it with `xl.writexl(db, 'updated.xlsx')`, if you are looking for dynamic updating then try `xlwings` library

**Reply by aspen33619** [reply]
Now, it works!!
I thought, the update was dynamic, because the documentation said, writexl does write the whole sheet. Now, that leads me to the problem that the rewrite cancels all my formatting. Therefore, either, I have to reformat by VBA macro or to import the written file into the formatted sheet. My next challenge.

Thanks again for your time.
